include/revng/Pypeline/ObjectID.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <set>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Model/TypeDefinition.h"
+#include "revng/Pipeline/Location.h"
+#include "revng/Pipes/Ranks.h"
+#include "revng/Support/MetaAddress.h"
+
+class ObjectID {
+private:
+  using TypeDefinitionKey = model::TypeDefinition::Key;
+  std::variant<std::monostate, MetaAddress, TypeDefinitionKey> Component;
```

Comment:
`Component` -> `Key`?

`Component` seems to imply there are multiples ones, but there aren't.